### DISCUSSION TITLE
[8일차 -윤경] 불변성이란?

### DISCUSSION THREAD

**Discussion by skyhanull** [zzangCS/zzangInterview#29] [프론트 면접 질문]
js에서 불변성이란 무엇인가?

**Comment by skyhanull** [comment]
js에서 불변성이란 변경할 수 없는 값을 말하는데 원시타입(boolean, null, undefined,number, string,symbol..등)이 있다

원시타입 이외의 모든 값은 객체타입이며 객체타입은 변경 가능한 값이다.

**Comment by ez1n** [comment]
**불변성**
    
⇒ 객체가 생성된 이후 상태를 변경할 수 없는 것

 원시타입 (불리언, 문자열, 숫자형, null, undefined, 심볼)은 불변성에 의해 변수에 할당할 때 완전히 새로운 값이 만들어져 재할당 된다.
 때문에 선언된 변수 원시값을 재할당 하는 경우 값 자체가 바뀌는 것이 아니라 새로운 주소 공간에 값이 할당되고 변수가 가리키는 주소가 변하게 되는 것이다.
